hadoop-common-project/hadoop-common/src/test/java/org/apache/hadoop/util/TestReflectionUtils.java

```diff
@@ -168,6 +168,13 @@ public void testLogThreadInfo() throws Exception {
         containsString("Process Thread Dump: " + title));
   }
 
+  @Test
+  public void testNewInstanceForNonDefaultConstructor() {
+    Object x = ReflectionUtils.newInstance(
+        NoDefaultCtor.class, null, new Class[] {int.class}, 1);
+    Assert.assertTrue(x instanceof NoDefaultCtor);
```

Comment:
I think `Assert` is not imported and compilation will fail. Also can we test the cases when number of parameter and argument doesn't match?

Reply:
Thanks for check this for me, done